core/services/beacon/settings.py

```diff
@@ -0,0 +1,117 @@
+import json
+import re
+import socket
+from typing import Any, Dict
+
+import psutil
+from commonwealth.settings import settings
+from pykson import (
+    IntegerField,
+    JsonObject,
+    ListField,
+    MultipleChoiceStringField,
+    ObjectListField,
+    StringField,
+)
+
+
+class InvalidIpAddress(Exception):
+    pass
+
+
+class DefaultSettings(JsonObject):
+    domain_names = ListField(item_type=str)
+    advertise = ListField(item_type=str)
+    ip = StringField()
+
+
+class Interface(JsonObject):
+    name = StringField()
+    domain_names = ListField(item_type=str)
+    advertise = ListField(item_type=str)
+    ip = StringField()
+
+    def get_phys(self) -> psutil._common.snicaddr:
+        """
+        return Physical interface from psutil
+        """
+        return psutil.net_if_addrs()[self.name]
+
+    def get_ip_str(self) -> str:
+        """
+        get ip as a string. this also translates 'ips[n]' to the n-th ip in that interface
+        """
+        try:
+            # validate using socket
+            socket.inet_aton(self.ip)
+            return str(self.ip)
+        except OSError:
+            pass
+
+        index = 0
+        matches = re.findall(r"ips\[(\d+)]", self.ip)
+        if len(matches) != 0:
+            index = int(matches[0])
+        address = str(self.get_phys()[index].address)
+        if address.count(":") > 0:
+            # this interface returned the mac address instead of ip, abort
+            raise InvalidIpAddress("Got a mac address instead of ip address")
+        return str(address)
```

Comment:
You can simplify the logic and use a general check with `inet_aton`
```suggestion        
        address = str(self.ip)
        # Check for 'ips[n]'
        matches = re.findall(r"ips\[(\d+)]", self.ip)
        if len(matches) != 0:
            index = int(matches[0])
            address = str(self.get_phys()[index].address)
        # Validate ip
        try:
            socket.inet_aton(address)
            return address
        except OSError:
            raise InvalidIpAddress(f"Invalid IP address: {address}")
        return address
```